common/membership/hashring.go

```diff
@@ -292,3 +292,11 @@ func (r *ring) compareMembers(members []HostInfo) (map[string]HostInfo, bool) {
 	}
 	return newMembersMap, changed
 }
+
+func castToMembers[T membership.Member](members []T) []membership.Member {
+	result := make([]membership.Member, 0, len(members))
+	for _, h := range members {
+		result = append(result, h)
+	}
+	return result
+}
```

Comment:
go generics make this such a weird-looking operation :|

any idea how often we do this kind of "list of things compatible with T" -> "[]T list" conversion?  seems like it might be worth making a common helper for it...

Reply:
hmmm.  seems regrettably not possible with go's generics 🤔.  would work fine if we could have it compile-time checked only on types used, not on all possible types, but we don't have that option...